modules/core/src/main/scala/fs2/kafka/internal/syntax.scala

```diff
@@ -186,34 +188,23 @@ private[kafka] object syntax {
   implicit final class KafkaFutureSyntax[A](
     private val future: KafkaFuture[A]
```

Comment:
Just for clarification: is `KafkaFuture` eager or lazy? If it's eager, so how `kafkaFuture.cancelable[F]` affects referential transparency? Maybe we should use the same approach is for scala futures?

Reply:
Sorry, just saw this - it's eager. I've pushed a change so that creation is suspended in a functional effect - is that what you meant?

Reply:
Short answer: before my change, `foo.cancelable[F]` wasn't referentially transparent. Fortunately this is an internal API only used in one place, so easy to change.